### DISCUSSION TITLE
Bounding rectangle constrained to parallel to a given axis.

### DISCUSSION THREAD

**Discussion by ghost** [shapely/shapely#1350] [Q&A] [answered]
Hi! Wonderful library, thanks!

I was wondering if given a Polygon and an axis, it would be possible to found the bounding rectangle constrained to be parallel to the given axis as shown in the image below:
![image](https://user-images.githubusercontent.com/85804972/159813620-92b9756f-df31-4225-8c27-a27d9839988d.png)

Thanks!

**Comment by mwtoews** [comment, marked as answer]
A similar feature is [`minimum_rotated_rectangle`](https://shapely.readthedocs.io/en/latest/manual.html#object.minimum_rotated_rectangle), but there is no built-in method to generate this for an arbitrary rotation.

Here's an idea: [rotate](https://shapely.readthedocs.io/en/latest/manual.html#shapely.affinity.rotate) the geometry by the axis angle, use the bounds to create a box geometry, then rotate it again by the opposite angle.
```python
from shapely import affinity, wkt
from shapely.geometry import box

def rotated_bbox(geom, angle):
    b0 = box(*affinity.rotate(geom, -angle, (0, 0)).bounds)
    return affinity.rotate(b0, angle, (0, 0))

g = wkt.loads("POLYGON ((140 160, 237 235, 300 190, 272 125, 160 130, 140 160))")
b = rotated_bbox(g, 10)
```

**Reply by ghost** [reply]
Thanks for the solution! Much appreciated!
